File: 2022/Day4/python/Day4.py
with open(r'../input.txt') as f:
    data = f.read().splitlines()
    data = [x.strip() for x in data]
        
data = [line.split(',') for line in data]
envelops, overlaps = 0,0

for pair in data:
    line = []
    for assn in pair:
        assn = assn.split('-')
        line.extend([int(string) for string in assn])
    
    if line[0] <= line[2] and line[1] >= line[3]:
        envelops += 1
    elif line[0] >= line[2] and line[1] <= line[3]:
        envelops += 1
        
    if line[1] >= line[2] and line[0] <= line[3]:
        overlaps += 1
    

print(f'Part 1: {envelops}')
print(f'Part 2: {overlaps}')

File: 2022/Day4/javascript/Day4.js
// Read File ../input.txt

// Part 1 and Part 2


// ZhengLinLei

const fs = require('fs');

// Read File Async
fs.readFile('../input.txt', 'utf8', (err, data) => {
    let sum = 0; sum2 = 0;
    data.split('\n').map(x => x.split(',')).map(x => x.map(y => y.split('-').map(Number))).forEach(i => {
        if((i[0][0] <= i[1][0] && i[0][1] >= i[1][1]) || (i[0][0] >= i[1][0] && i[0][1] <= i[1][1])) sum ++;
        if((i[0][1] >= i[1][0] && i[0][0] <= i[1][1])) sum2 ++;
    });


    

    console.log("Part 1: ", sum);
    console.log("Part 2: ", sum2);
});
